main/main.cpp

```diff
@@ -1938,12 +1938,13 @@ Error Main::setup2(Thread::ID p_main_tid_override) {
 #ifdef TOOLS_ENABLED
 	if (editor || project_manager || cmdline_tool) {
 		EditorPaths::create();
-		if (EditorPaths::get_singleton()->is_self_contained()) {
-			if (ProjectSettings::get_singleton()->get_resource_path() == OS::get_singleton()->get_executable_path().get_base_dir()) {
-				ERR_PRINT("You are trying to run a self-contained editor at the same location as a project. This is not allowed, since editor files will mix with project files.");
-				OS::get_singleton()->set_exit_code(EXIT_FAILURE);
-				return FAILED;
-			}
+	}
+	if ((editor || cmdline_tool) && EditorPaths::get_singleton()->is_self_contained()) {
```

Comment:
I'm not sure this is enough as for example this would break when running `godot --doctool` when self-contained.

I think what you should discriminate on is `found_project`, which you'd need to make global from the previous `setup()`. That's basically what tells you that you're running with a valid project path and loading the `project.godot`.

Reply:
Are users running doctool in self-contained mode?

Originally I had this code only inside of the export code, maybe I should move it back there?

Reply:
Updated the PR to use `found_project`, indeed this is better and I did some testing and it works as expected.